homeassistant/components/comfoconnect/config_flow.py

```diff
@@ -0,0 +1,158 @@
+    bridges = await hass.async_add_executor_job(Bridge.discover, host)
+    if not bridges:
+        raise ConnectionError(f"No bridges found at host {host}")
+    ccb = ComfoConnectBridge(
+        hass,
+        bridge=bridges[0],
+        name=input_data.get(CONF_NAME, DEFAULT_NAME),
+        token=input_data.get(CONF_TOKEN, DEFAULT_TOKEN),
+        friendly_name=input_data.get(CONF_USER_AGENT, DEFAULT_USER_AGENT),
+        pin=input_data.get(CONF_PIN, DEFAULT_PIN),
+    )
+    try:
+        await ccb.connect()
+    finally:
+        await ccb.disconnect()
+    return ccb.unique_id
+
+
+class ComfoConnectConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """ConfigFlow for ComfoConnect integration."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        bridges = await self.hass.async_add_executor_job(Bridge.discover)
+        if bridges:
+            # one or more bridges auto-discovered
+            host_field_val = vol.All(str, vol.In([bridge.host for bridge in bridges]))
+            host_field_key = vol.Required(CONF_HOST, default=bridges[0].host)
+        else:
+            # no bridge discovered
+            host_field_val = str
+            host_field_key = vol.Required(CONF_HOST)
+        schema = vol.Schema(
+            {
+                host_field_key: host_field_val,
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                vol.Optional(CONF_PIN, default=DEFAULT_PIN): int,
+            }
+        )
+        if user_input is not None:
+            unique_id = None
+            try:
+                unique_id = await validate_input(self.hass, user_input)
+            except ConnectionError:
+                errors[CONF_HOST] = "cannot_connect"
+            except PyComfoConnectNotAllowed:
+                errors[CONF_PIN] = "invalid_auth"
+            # We have to catch a very broad "Exception" here, since that's what
+            # pycomfoconnect raises
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            if unique_id:
+                await self.async_set_unique_id(unique_id)
+                self._abort_if_unique_id_configured()
+                user_input[CONF_TOKEN] = secrets.token_hex(16)
+                user_input[CONF_USER_AGENT] = DEFAULT_USER_AGENT
+                self.context[STEP_ID_USER] = user_input
+                return await self.async_step_sensor_selection()
+        return self.async_show_form(
+            step_id=STEP_ID_USER, data_schema=schema, errors=errors
+        )
+
+    async def async_step_sensor_selection(self, user_input=None):
+        """Handle sensor selection step."""
+        schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SENSORS,
+                ): cv.multi_select({sensor.key: sensor.name for sensor in SENSOR_TYPES})
+            }
+        )
```

Comment:
We should not offer this selection (especially not in the configuration flow). Instead, add them all and let the user decide.

Reply:
@frenck Adding them all used to give a performance impact on the bridge (although I haven't tested this recently), so it's best to only subscribe for the sensors you need. 

But maybe we should select a set of "minimal defaults"?

Reply:
> Adding them all used to give a performance impact on the bridge (although I haven't tested this recently), so it's best to only subscribe for the sensors you need.

In that case, it still should not be done this way. Instead, we can disable the less common sensors by default in the entity registry.